### PR TITLE
Enable tracing bp model

### PR DESCRIPTION
**Pull request by pprajapatiTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/28919

### Problem description
1. Add tracing support in demo. use --enable-trace
2. Some clean ups (like fixing PP model references)

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by pprajapatiTT** [comment]
CI: https://github.com/tenstorrent/tt-metal/actions/runs/19125352361

File: models/demos/deepseek_v3/tt/generator.py [hunk 1]
```diff
@@ -681,5 +693,92 @@ def _prefill(self, tokens: torch.Tensor, user_id: int) -> torch.Tensor:
         ttnn.deallocate(logits_tt)
         return logits  # [1, 1, seq_len, V]
 
+    def _capture_decode_trace(
+        self, init_tokens: torch.Tensor, positions: torch.Tensor, batch_size_per_row: int
+    ) -> None:
+        """Allocate persistent inputs, capture trace for one decode iteration, and store trace state."""
+        assert self._trace_id is None, "Trace already captured"
+
+        # 1) Warm-up compile run (no trace) to keep compilation out of capture
```
Comment (yieldthought) [inline models/demos/deepseek_v3/tt/generator.py:702]: I don't think this should be necessary - tracing only captures the device behaviour and not anything that happens on the host. Were there problems when running without this step?

**Review by yieldthought — APPROVED** [review]

File: models/demos/deepseek_v3/tt/generator.py (hunk 1 above)
Comment (pprajapatiTT) [reply models/demos/deepseek_v3/tt/generator.py:702]: I hit this assert without warm-up

`2025-11-06 23:27:52.837 | critical |          Always | Writes are not supported during trace capture. trace id: StrongType<unsigned int, tt::tt_metal::distributed::MeshTraceIdTag>(value=0) (assert.hpp:103)`
